src/de/grobox/blitzmail/NoteActivity.java

```diff
@@ -86,6 +90,28 @@ public void onCancel(DialogInterface dialog) {
 		dialog.setCanceledOnTouchOutside(false);
 		dialog.show();
 
+		// TODO. I seem to be able to get hold of clipboard contents,
+		// but setText doesn't seem to work.
+
+		// ClipboardManager cm = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+		// ClipData cd = cm.getPrimaryClip();
+		// if (cd != null) {
+		// 	ClipData.Item item = cd.getItemAt(0);
+		// 	textView.setText(item.coerceToText(this).toString());
+		// }
+
+		// So, this is a separate setText() attempt, which doesn't
+		// work. I have tried getting hold of edittext-view from mView,
+		// and also from dialog. No difference. Also doing it before
+		// and after the build.create() call, to no avail. Using the
+		// textView member set in the beginning of this function
+		// doesn't work either. We do get hold of the edittext-view
+		// here, but setText() on it does nothing.
+		EditText t = (EditText) mView.findViewById(R.id.text);
```

Comment:
mView is not the dialog which is not included in that view hierarchy. You need to get the view of the dialog instead.

Reply:
Hmm we are already doing `textView = (TextView) mView.findViewById(R.id.text);` above and interact with it in onResume as well, so that should work... Maybe your text gets overwritten in onResume()?

Reply:
Aha, the resume does that! Always. It seems like a `note` string with length 0 is stored, have to avoid picking up such I think

Reply:
Maybe this can be solved by calling `saveText(null);` instead of `saveText("");` so resume only changes the text when it is not null